beetsplug/discogs.py

```diff
@@ -205,11 +205,27 @@ def get_albums(self, query):
             self._log.debug(u"Communication error while searching for {0!r}",
                             query, exc_info=True)
             return []
-        return [self.get_album_info(release) for release in releases[:5]]
+        return filter(None, [self.get_album_info(release)
+                             for release in releases[:5]])
 
     def get_album_info(self, result):
         """Returns an AlbumInfo object for a discogs Release object.
         """
+        # Explicitly reload the `Release` fields, as they might not be yet
+        # present if the result is from a `discogs_client.search()`.
+        if not result.data.get('artists'):
+            result.refresh()
```

Comment:
This is included in order to be able to perform the `if not all` check when the function is called from `get_albums()`, as all the required info is not included in that case. The `refresh()` was [implicitly called](https://github.com/discogs/discogs_client/blob/master/discogs_client/models.py#L245) when `result.artists` was invoked at L229:
```artist, artist_id = self.get_artist([a.data for a in result.artists])```